webpack/assets/javascripts/react_app/components/common/Slot/Slot.js

```diff
@@ -1,8 +1,25 @@
-import { cloneElement, isValidElement } from 'react';
+import { cloneElement, isValidElement, useState } from 'react';
 import PropTypes from 'prop-types';
 
-const Slot = ({ fills, id, multi, children = null, ...props }) => {
+const Slot = ({
+  fills,
+  id,
+  multi,
+  children = null,
+  deprecated,
+  replacedBy,
+  ...props
+}) => {
+  const [warned, setWarned] = useState(false);
   const addProps = object => {
+    if (deprecated && fills?.length && !warned) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        `Slot with id '${id}' is deprecated and will be removed in the next release. Please use '${replacedBy}' instead.`
```

Comment:
This is a great addition so we won't break plugins!
Should it specify the next release version number?

Reply:
i'm not opposed! Would that be 3.3 or 3.4?

Reply:
Lets remove it in 3.4